readthedocs/embed/v3/views.py

```diff
@@ -0,0 +1,345 @@
+            'html',
+            version_slug=version.slug,
+            include_file=False,
+            version_type=version.type,
+        )
+        file_path = build_media_storage.join(
+            storage_path,
+            filename,
+        )
+        try:
+            with build_media_storage.open(file_path) as fd:  # pylint: disable=invalid-name
+                return fd.read()
+        except Exception:  # noqa
+            log.warning('Unable to read file. file_path=%s', file_path)
+
+        return None
+
+    def _get_content_by_fragment(self, url, fragment, external, doctool, doctoolversion):
+        if external:
+            page_content = self._download_page_content(url)
+        else:
+            project = self.unresolved_url.project
+            version_slug = self.unresolved_url.version_slug
+            filename = self.unresolved_url.filename
+            page_content = self._get_page_content_from_storage(project, version_slug, filename)
+
+        return self._parse_based_on_doctool(page_content, fragment, doctool, doctoolversion)
+
+    def _find_main_node(self, html):
+        main_node = html.css_first('[role=main]')
+        if main_node:
+            log.info('Main node found. selector=[role=main]')
+            return main_node
+
+        main_node = html.css_first('main')
+        if main_node:
+            log.info('Main node found. selector=main')
+            return main_node
+
+        first_header = html.body.css_first('h1')
+        if first_header:
+            log.info('Main node found. selector=h1')
+            return first_header.parent
+
+    def _parse_based_on_doctool(self, page_content, fragment, doctool, doctoolversion):
+        # pylint: disable=unused-argument
+        if not page_content:
+            return
+
+        node = None
+        if fragment:
+            selector = f'#{fragment}'
+            node = HTMLParser(page_content).css_first(selector)
+        else:
+            html = HTMLParser(page_content)
+            node = self._find_main_node(html)
+
+        if not node:
+            return
+
+        if doctool == 'sphinx':
+            # Handle ``dt`` special cases
+            if node.tag == 'dt':
+                if 'glossary' in node.parent.attributes.get('class'):
+                    # Sphinx HTML structure for term glossary puts the ``id`` in the
+                    # ``dt`` element with the title of the term. In this case, we
+                    # return the parent node which contains the definition list
+                    # and remove all ``dt/dd`` that are not the requested one
+
+                    # Structure:
+                    # <dl class="glossary docutils">
+                    # <dt id="term-definition">definition</dt>
+                    # <dd>Text definition for the term</dd>
+                    # ...
+                    # </dl>
+
+                    # TODO: figure it out if it's needed to remove the siblings here
```

Comment:
are we going to resolve these todos in this PR?

Reply:
It's not a blocker to move forward.